src/pybind/mgr/dashboard_v2/controllers/rgw.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import json
+
+from ..tools import ApiController, RESTController
+from .. import logger
+
+
+@ApiController('rgw')
+class Rgw(RESTController):
+
+    def __init__(self):
+        self.daemon = RgwDaemon()
+
+
+@ApiController()
+class RgwDaemon(RESTController):
+
+    def list(self):
+        daemons = []
+        for server in self.mgr.list_servers():
+            for service in server['services']:
+                if service['type'] == 'rgw':
+                    metadata = self.mgr.get_metadata('rgw', service['id'])
+                    status = self.mgr.get_daemon_status('rgw', service['id'])
+                    if 'json' in status:
+                        try:
+                            status = json.loads(status['json'])
+                        except ValueError:
+                            msg = "{0} had invalid status json"
+                            logger.warning(msg.format(service['id']))
+                            status = {}
+                    else:
+                        msg = '{0} has no key "json" in status'
+                        logger.warning(msg.format(service['id']))
+
+                    # extract per-daemon service data and health
+                    daemon = {
+                        'id': service['id'],
+                        'version': metadata['ceph_version'],
+                        'server_hostname': server['hostname'],
+                        'service': service,
+                        'server': server,
+                        'metadata': metadata,
+                        'status': status,
+                        'url': "{0}/api/rgw/daemon/{1}".format(
+                            self.mgr.url_prefix, service['id'])
+                    }
+
+                    daemons.append(daemon)
+
+        return {'daemons': sorted(daemons, key=lambda k: k['id'])}
```

Comment:
I think ``return sorted(daemons, key=lambda k: k['id'])`` is enough here. We do not need to adapt the structure of the returned data from the origin dashboard.

Reply:
This is exactly the structure from the original dashboard. But note that I slightly (IMO) improved the structure of the detail view.

Reply:
Hmmm, but if i request the daemons via api/deamon i do not expect a return object that is a dict with a key ``daemons``. This is somewhat doubled.

Reply:
If this will return the entire list of daemons, I would prefer it to simply return the list of daemons.

Only in situations where we have server side pagination and filtering, we would need a object as the return value. And I think in oA we return the list on a 'result' property.